### ISSUE TITLE
We should have a custom 404 page on clayui.com

### ISSUE THREAD

**Issue by kresimir-coko** [liferay/clay#3181]
## What is your proposal?
I propose that we have a dedicated 404 page on clayui.com

## Why would adopting this proposal be beneficial?
It would display an insightful error to our users instead of the Gatsby's default 404 page which is pretty bad.

## What are the alternatives to this proposal?
Not having a custom 404 page.

**Comment by bryceosterhaus** [comment]
Sounds good to me! Feel free to get creative with it.

For inspiration, we did something like this on Loop.
![Screen Shot 2020-04-28 at 10 05 17 AM](https://user-images.githubusercontent.com/6843530/80516063-c6383380-8937-11ea-98a3-b1c5abd58e9d.png)
